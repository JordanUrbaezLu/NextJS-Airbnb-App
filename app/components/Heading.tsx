"use client";

interface HeadingProps {
  title: string;
  subtitle?: string;
  center?: boolean;
}

const Heading: React.FC<HeadingProps> = ({ title, subtitle, center }) => {
  return (
    <div
      aria-level={1}
      className={center ? "text-center" : "text-start"}
      role="heading"
    >
      <div className="text-2xl font-bold">{title}</div>
      <div className="mt-2 font-light text-neutral-500">{subtitle}</div>
    </div>
  );
};

export default Heading;
