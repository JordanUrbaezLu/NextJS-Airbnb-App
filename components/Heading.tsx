"use client";

interface HeadingProps {
  title: string;
  subtitle?: string;
  center?: boolean;
}

const Heading: React.FC<HeadingProps> = ({ title, subtitle, center }) => {
  return (
    <div
      aria-level={1}
      className={center ? "text-center" : "text-start"}
      role="heading"
    >
      <div className="text-2xl font-bold text-black dark:text-white">
        {title}
      </div>
      <div className="mt-2 font-light text-neutral-500 dark:text-neutral-100">
        {subtitle}
      </div>
    </div>
  );
};

export default Heading;
