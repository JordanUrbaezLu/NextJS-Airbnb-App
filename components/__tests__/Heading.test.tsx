import { render, screen } from "@testing-library/react";
import Heading from "../Heading";

describe("Heading", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<Heading title="Airbnb" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("Props", () => {
    test("Should render center prop correctly", () => {
      render(<Heading title="Airbnb" center />);

      expect(screen.getByRole("heading")).toHaveClass("text-center");
    });
  });
});
