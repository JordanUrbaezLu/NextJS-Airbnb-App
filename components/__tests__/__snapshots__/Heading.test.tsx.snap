// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Should render component correctly. 1`] = `
<div
  aria-level="1"
  class="text-start"
  role="heading"
>
  <div
    class="text-2xl font-bold text-black dark:text-white"
  >
    Airbnb
  </div>
  <div
    class="mt-2 font-light text-neutral-500 dark:text-neutral-100"
  />
</div>
`;
